pallets/regulated-assets/src/lib.rs

```diff
@@ -0,0 +1,283 @@
+        #[pallet::weight(<T as Config>::WeightInfo::issue_sbt())]
+        pub fn issue_sbt(
+            origin: OriginFor<T>,
+            symbol: AssetSymbol,
+            name: AssetName,
+            initial_supply: Balance,
+            allowed_assets: Vec<AssetIdOf<T>>,
+            description: Option<Description>,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            // Check permission `who` can issue SBT
+            Permissions::<T>::check_permission(who.clone(), ISSUE_SBT)?;
+
+            let sbt_asset_id = T::AssetManager::register_from(
+                &who,
+                symbol,
+                name.clone(),
+                DEFAULT_BALANCE_PRECISION,
+                initial_supply,
+                true,
+                None,
+                description.clone(),
+            )?;
+            let metadata = SoulboundTokenMetadata {
+                name,
+                description,
+                allowed_assets: allowed_assets.clone(),
+            };
+            <SoulboundAsset<T>>::insert(sbt_asset_id, &metadata);
+
+            for allowed_asset in allowed_assets {
+                <SBTsByAsset<T>>::mutate(allowed_asset, |sbts| {
+                    sbts.insert(sbt_asset_id);
+                });
+            }
+
+            Self::deposit_event(Event::SoulboundTokenIssued {
+                asset_id: sbt_asset_id,
+                owner: who,
+                metadata,
+            });
+
+            Ok(())
+        }
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        AssetRegulated {
+            asset_id: AssetIdOf<T>,
+        },
+        SoulboundTokenIssued {
+            asset_id: AssetIdOf<T>,
+            owner: AccountIdOf<T>,
+            metadata: SoulboundTokenMetadata<AssetIdOf<T>>,
+        },
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        SoulboundAssetNotOperationable,
+        SoulboundAssetNotTransferable,
+        OnlyAssetOwnerCanRegulate,
+        AssetAlreadyRegulated,
+        AllInvolvedUsersShouldHoldSBT,
+    }
+
+    #[pallet::type_value]
+    pub fn DefaultRegulatedAsset<T: Config>() -> bool {
+        false
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn regulated_asset)]
+    pub type RegulatedAsset<T: Config> =
+        StorageMap<_, Identity, AssetIdOf<T>, bool, ValueQuery, DefaultRegulatedAsset<T>>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn soulbound_asset)]
+    pub type SoulboundAsset<T: Config> =
+        StorageMap<_, Identity, AssetIdOf<T>, SoulboundTokenMetadata<AssetIdOf<T>>, OptionQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn sbts_by_asset)]
+    pub type SBTsByAsset<T: Config> =
+        StorageMap<_, Identity, AssetIdOf<T>, BTreeSet<AssetIdOf<T>>, ValueQuery>;
+}
+
+impl<T: Config> AssetRegulator<AccountIdOf<T>, AssetIdOf<T>> for Pallet<T> {
+    fn check_permission(
```

Comment:
I don't where it is used. You use `check_permission` from Permissions pallet.

Reply:
``` 
// Check permission `who` can issue SBT
Permissions::<T>::check_permission(who.clone(), ISSUE_SBT)?;
```

I used it to check if the issuer account has such permission to issue sbt.
I thought of that we should restrict this part to only regulated-agency's account.

What do you think?

Reply:
I'm asking about other thing.

You call `check_permission()` in the `issue_sbt` extrinsic by calling:
```rust
Permissions::<T>::check_permission(who.clone(), ISSUE_SBT)?;
```

This call goes into the `permission` pallet. And it executes `permission::check_permission()`.
It works well if you have assigned the `ISSUE_SBT` permission for the acount, but it doesn't call the `regulated-assets::check_permission()`.

The quistion is - where the current fn `regulated-assets::check_permission()` is used?

Reply:
- AssetRegulator is implemented for (A, B)
https://github.com/sora-xor/sora2-network/blob/1012-tech-pallet-for-regulated-assets-support/common/src/traits.rs#L1469

- AssetRegulator is used in assets pallet for mint, transfer, and burn functions to check permission
 The definition for Runtime should be like [this](https://github.com/sora-xor/sora2-network/blob/1012-tech-pallet-for-regulated-assets-support/pallets/regulated-assets/src/mock.rs#L92)
 
 @Alexey-N-Chernyshov @wer1st

Reply:
In this case you have to update `assets::Config` in `runtime/src/lib.rs` as well by the same way